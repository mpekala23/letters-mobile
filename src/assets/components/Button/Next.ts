const Next = `<svg width="100%" height="100%" viewBox="0 0 24 21" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M24 10.5L16.2772 21H12.1188L18.6535 12.25H0V8.75H18.6535L12.1188 0H16.2772L24 10.5Z" fill="white"/>
</svg>
`;

export default Next;
