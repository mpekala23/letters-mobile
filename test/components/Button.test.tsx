import React from 'react';
import { Button } from '@components';
import { fireEvent, render, toJSON, act } from '@testing-library/react-native';
import { Colors } from '@styles';
import { sleep } from '@utils';
import Next from '@assets/components/Button/Next';

const setup = (propOverrides = {}) => {
  const props = {
    onPress: jest.fn(),
    buttonText: 'press me',
    ...propOverrides,
  };
  return {
    ...render(<Button {...props} />),
    props,
  };
};

describe('Button component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should implement style props when enabled', () => {
    const containerStyle = { backgroundColor: 'green' };
    const textStyle = { color: 'red' };
    const { getByText, getByTestId } = setup({
      containerStyle,
      textStyle,
    });
    expect(getByTestId('clickable').props.style[2]).toEqual(containerStyle);
    expect(getByText('press me').props.style[3]).toEqual(textStyle);
  });

  it('should implement a reverse style', () => {
    const { getByText } = setup({
      reverse: true,
    });
    expect(getByText('press me').props.style[1].color).toEqual(Colors.PINK_500);
  });

  it('should implement a disabled style', () => {
    const disabledContainerStyle = { backgroundColor: 'green' };
    const disabledTextStyle = { color: 'red' };
    const { getByText, getByTestId } = setup({
      disabledContainerStyle,
      disabledTextStyle,
      enabled: false,
    });
    expect(getByTestId('clickable').props.style[3]).toEqual(
      disabledContainerStyle
    );
    expect(getByText('press me').props.style[4]).toEqual(disabledTextStyle);
  });

  it('should implement a link style', () => {
    const containerStyle = { backgroundColor: 'green' };
    const { getByTestId } = setup({
      containerStyle,
      link: true,
    });
    expect(getByTestId('clickable').props.style).toEqual(containerStyle);
  });

  it('should call onPress() when pressed and enabled', () => {
    const { props, getByTestId } = setup({
      enabled: true,
    });
    fireEvent.press(getByTestId('clickable'));
    // expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it('should not call onPress() when pressed and disabled', () => {
    const { props, getByTestId } = setup({
      enabled: false,
    });
    fireEvent.press(getByTestId('clickable'));
    // expect(props.onPress).toHaveBeenCalledTimes(0);
  });

  it('should show next arrow icon when prop is true', () => {
    const { getByTestId } = setup({
      showNextIcon: true,
    });
    expect(getByTestId('nextIcon').children[0].props.xml).toBe(Next);
  });
});
