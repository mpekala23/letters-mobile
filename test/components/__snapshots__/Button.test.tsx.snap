// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.7}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FF7171",
          "borderRadius": 15,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 5,
          "paddingHorizontal": 14,
        },
        Object {},
        Object {},
        Object {},
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2.62,
        },
      ]
    }
    testID="clickable"
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins-Medium",
          },
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "700",
            "padding": 6,
          },
          Object {},
          Object {},
          Object {},
        ]
      }
    >
      press me
    </Text>
  </TouchableOpacity>
</View>
`;
